### ISSUE TITLE
Upgrade of dependencies

### ISSUE THREAD

**Issue by dcoraboeuf** [nemerosa/ontrack#436]
Global upgrade of dependencies in Ontrack. Target Spring Boot 1.4.

**Comment by dcoraboeuf** [comment]
Targeting 2.25. Some commits will be picked from the 3.0 branch and the 3.0 branch will be deleted.

**Comment by dcoraboeuf** [comment]
(!) regression - the static web pages cannot be accessed any longer o_O

**Comment by dcoraboeuf** [comment]
See https://jira.spring.io/browse/SPR-14368 for the source issue. It is supposed to be fixed with Spring 4.3.2 but is still failing.

**Comment by dcoraboeuf** [comment]
In `WebConfig`, the `ContentNegotiationConfigurer` is set to _not favour_ the path extension:

```java
configurer.favorPathExtension(false).favorParameter(false);
```

Changed this to:

```java
configurer.favorParameter(false);
```

This fixes the issue of JS files being served in binary - but might introduce other regressions.
